extensions/panache/panache-mock/deployment/src/main/java/io/quarkus/panache/mock/deployment/PanacheMockResourceProcessor.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.panache.mock.deployment;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.panache.common.deployment.PanacheMethodCustomizerBuildItem;
+import io.quarkus.runtime.LaunchMode;
+
+public final class PanacheMockResourceProcessor {
```

Comment:
I wonder if we could / should ditch the full blown extension approach here, and just introduce a test module that utilizes the `TestBuildChainCustomizerProducer` we are introducing in #8491

Reply:
Good idea, done.

Reply:
Excellent